src/Utilities.js

```diff
@@ -0,0 +1,123 @@
+import * as CESymbols from './CustomElementInternalSymbols';
+
+const reservedTagList = new Set([
+  'annotation-xml',
+  'color-profile',
+  'font-face',
+  'font-face-src',
+  'font-face-uri',
+  'font-face-format',
+  'font-face-name',
+  'missing-glyph',
+]);
+
+/**
+ * @param {string} localName
+ * @returns {boolean}
+ */
+export function isValidCustomElementName(localName) {
+  const reserved = reservedTagList.has(localName);
+  const validForm = /^[a-z][.0-9_a-z]*-[\-.0-9_a-z]*$/.test(localName);
+  return !reserved && validForm;
+}
+
+/**
+ * @private
+ * @param {!Node} node
+ */
+export function isConnected(node) {
```

Comment:
`isConnected` is true as long as a node's shadow-including root is a document but I've also updated this to bail out early if we hit an import's root node (a document in native, an element which I've marked with a flag in the polyfill).